generator/templates/wire/WireCmd.h

```diff
@@ -15,10 +15,41 @@
 #ifndef WIRE_WIRECMD_AUTOGEN_H_
 #define WIRE_WIRECMD_AUTOGEN_H_
 
-#include <nxt/nxt.h>
+namespace nxt { namespace wire {
 
-namespace nxt {
-namespace wire {
+    using ObjectId = uint32_t;
+    using ObjectSerial = uint32_t;
+
+    enum class DeserializeResult {
+        Success,
+        FatalError,
+        ErrorObject,
+    };
+
+    // Interface to allocate more space to deserialize pointed-to data.
+    // nullptr is treated as an error.
+    class DeserializeAllocator {
+        public:
+            virtual char* GetSpace(size_t size) = 0;
```

Comment:
Nit: if you made this one void * as well, you could use static_cast instead of reinterpret_cast below.